### ISSUE TITLE
PGFunction.from_database fails when schema contains a procedure

### ISSUE THREAD

**Issue by captaincoordinates** [olirice/alembic_utils#10]
alembic_utils/pg_function.py's from_database method queries Postgres metadata for SQL definitions of existing functions to see if a PGFunction should be created or replaced in a migration. Unfortunately its SQL query is missing a critical clause and as a result it will attempt to parse a procedure as a PGFunction if a procedure exists in the schema.

Error (truncated):

>   File "/usr/local/lib/python3.7/site-packages/alembic_utils/pg_function.py", line 128, in from_database
>     db_functions = [PGFunction.from_sql(x[3]) for x in rows]
>   File "/usr/local/lib/python3.7/site-packages/alembic_utils/pg_function.py", line 128, in <listcomp>
>     db_functions = [PGFunction.from_sql(x[3]) for x in rows]
>   File "/usr/local/lib/python3.7/site-packages/alembic_utils/pg_function.py", line 42, in from_sql
>     raise SQLParseFailure(f'Failed to parse SQL into PGFunction """{sql}"""')
> alembic_utils.exceptions.SQLParseFailure: Failed to parse SQL into PGFunction """CREATE OR REPLACE PROCEDURE [...] """

I believe the solution for this issue is to update the SQL query in from_database with an additional clause `and p.prokind = 'f'`, i.e.
```
        with extension_functions as (
            select
                objid as extension_function_oid
            from
                pg_depend
            where
                -- depends on an extension
                deptype='e'
                -- is a proc/function
                and classid = 'pg_proc'::regclass
        )

        select
            n.nspname as function_schema,
            p.proname as function_name,
            pg_get_function_arguments(p.oid) as function_arguments,
            case
                when l.lanname = 'internal' then p.prosrc
                else pg_get_functiondef(p.oid)
            end as create_statement,
            t.typname as return_type,
            l.lanname as function_language
        from
            pg_proc p
            left join pg_namespace n on p.pronamespace = n.oid
            left join pg_language l on p.prolang = l.oid
            left join pg_type t on t.oid = p.prorettype
            left join extension_functions ef on p.oid = ef.extension_function_oid
        where
            n.nspname not in ('pg_catalog', 'information_schema')
            -- Filter out functions from extensions
            and ef.extension_function_oid is null
            and n.nspname::text = '{schema}'
            and p.prokind = 'f';
```

**Comment by olirice** [comment]
Thanks for reporting the issue! 

The docs for pg_proc agree
https://www.postgresql.org/docs/11/catalog-pg-proc.html

If you open a PR I’ll get it merged and released on pypi asap

Otherwise I’ll take a look over the weekend

**Comment by olirice** [comment]
The update is available on pypi in release 0.2.8

**Comment by captaincoordinates** [comment]
Wow that was fast, much appreciated! I was planning to post a PR this morning but you beat me to it.

I'm looking at what's involved to also support PGProcedure and PGTrigger and at some point over the next week or two I might create a feature request and a PR for those
